bayes_opt/bayesian_optimization.py

```diff
@@ -137,6 +136,14 @@ def __init__(self,
         self._allow_duplicate_points = allow_duplicate_points
         self._queue = Queue()
 
+        if acquisition_function is None:
```

Comment:
Do we want to use a default here, or do we want to throw an error - that is, if people adapt the new version, we force them to update their codes? I feel like we got ourselves in trouble in the past by silently using defaults?

Reply:
The idea was to mirror [this line](https://github.com/bayesian-optimization/BayesianOptimization/blob/f2c720e76c6de2d5b334a3e971acdbc28ab37ed9/bayes_opt/bayesian_optimization.py#L304) in `.maximize`. So for someone not supplying any acquisition function, the behaviour should remain the same. On the other hand, if someone is used to supply an acquisition function to `.maximize` or `.suggest`, it would now crash with a `TypeError: got an unexpected keyword argument`.

Reply:
there's also the option of adding an `acquisition_function` argument to `maximize` and explicitly raise an error if someone supplies it.

Reply:
I think the current behavior is probably good, I like that it throws an error in more complex cases.